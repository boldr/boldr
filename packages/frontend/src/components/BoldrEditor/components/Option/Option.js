/* eslint-disable react/no-unused-prop-types */
/* @flow */

import * as React from 'react';
import { OptionWrapper } from './Option.styled';

export type Props = {
  onClick: Function,
  children: React.ChildrenArray<React.Node>,
  value?: string,
  active?: boolean,
  className?: string,
  isDark: boolean,
  disabled?: boolean,
  title?: string,
};

export default class Option extends React.PureComponent<Props, *> {
  static displayName = 'Option';
  static defaultProps = {
    disabled: false,
    isDark: false,
  };

  props: Props;

  onClick: Function = () => {
    const { disabled, onClick, value } = this.props;
    if (!disabled) {
      onClick(value);
    }
  };

  render() {
    const { children, className, active, disabled, title, isDark } = this.props;
    return (
      <OptionWrapper
        className={className}
        onClick={this.onClick}
        isDark={isDark}
        aria-selected={active}
        active={active}
        disabled={disabled}
        title={title}
      >
        {children}
      </OptionWrapper>
    );
  }
}
