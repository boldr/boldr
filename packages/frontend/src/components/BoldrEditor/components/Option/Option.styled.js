/* eslint-disable no-unused-vars */
import * as React from 'react';
import styled from 'styled-components';

export const OptionWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  min-width: 25px;
  height: 20px;
  padding: 3px;
  background-color: transparent;
  border: ${props => (props.isDark ? '1px solid #243140' : '1px solid #fafafa')};
  border-radius: 2px;
  margin: 0 4px;
  text-transform: capitalize;
  cursor: pointer;
  svg {
    color: ${props => (props.active ? '#00bcd4!important' : 'inherit')};
  }
  &:hover {
    svg {
      color: #00bcd4 !important;
    }
    opacity: ${props => (props.disabled ? 0.3 : 'inherit')};
    cursor: ${props => (props.disabled ? 'default' : 'inherit')};
  }
`;
