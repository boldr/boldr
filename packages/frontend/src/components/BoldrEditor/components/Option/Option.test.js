import React from 'react';
import { spy } from 'sinon';
import { shallowToJson } from 'enzyme-to-json';
import 'jest-styled-components';
import { shallow, mount } from 'enzyme';
import Option from './Option';

describe('<Option />', () => {
  it('should render in a div', () => {
    const wrapper = shallow(
      <Option value="f" onClick={() => {}}>
        <span>abcd</span>
      </Option>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should have child element passed after mount', () => {
    const option = mount(
      <Option value="f" onClick={() => {}}>
        <span>abcd</span>
      </Option>,
    );
    expect(option.children().length).toEqual(1);
    expect(option.children().type()).toEqual('span');
  });

  it('should execute funcion passed in onClick props when clicked', () => {
    const onClick = spy();
    const option = mount(
      <Option value="b" onClick={onClick}>
        <span>hello</span>
      </Option>,
    );
    option.children().simulate('click');
    expect(onClick.calledOnce).toEqual(true);
  });

  it('should not execute funcion passed in onClick props when clicked if disabled', () => {
    const onClick = spy();
    const option = mount(
      <Option value="b" onClick={onClick} disabled>
        <span>hello</span>
      </Option>,
    );
    option.children().simulate('click');
    expect(onClick.called).toEqual(false);
  });
});
