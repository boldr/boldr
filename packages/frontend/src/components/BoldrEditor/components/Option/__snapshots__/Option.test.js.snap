// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Option /> should render in a div 1`] = `
<Option.styled__OptionWrapper
  disabled={false}
  isDark={false}
  onClick={[Function]}
>
  <span>
    abcd
  </span>
</Option.styled__OptionWrapper>
`;
